### ISSUE TITLE
Cloudflare causes async request timeout 

### ISSUE THREAD

**Issue by jacobmasters** [Prospress/action-scheduler#826]
Had a lot of async requests fails on subscription payments, all stating timeouts, but with nothing in the error log to suggest a timeout/script error, so I disabled Cloudflare on the domain and async requests are working again, any idea if there is a url to whitelist, or a pattern to match to stop this or route Cloudflare around it?

scheduled action 871646 (subscription payment) failed to finish processing after 300 seconds

**Comment by barryhughes** [comment]
By default, async requests from Action Scheduler contain `async_request_queue_runner` in the URL query, so perhaps you can match against that and instruct CloudFlare to allow these requests? Alternatively, it's possible to customize async queue runner requests and modify the User Agent string (and various other things, too—link explaining this in more detail below).

Yet another approach (how viable this is depends on your hosting environment and how much control you can exercise there) would be stop using the async queue runner altogether, and move toward an exclusively WP CLI-based queue running strategy.

Links/resources:

- [Modifying async queue runner requests](https://actionscheduler.org/faq/#how-can-i-change-the-action-scheduler-user-agent-to-better-identify-its-requests)
- [Running Action Scheduler via WP CLI](https://actionscheduler.org/wp-cli/)
- [Tweaking performance/scalability](https://actionscheduler.org/perf/)

**Comment by jacobmasters** [comment]
Does running the scheduler via CLI still log fails and responses in the normal way?

Also the cloudflare instruction may work, but to use there "Rules" and disable specific settings you need to specify at least the start of the url, with domain, so http://example.com/wp-admin/* any more specifics on the post url?

**Comment by barryhughes** [comment]
> Does running the scheduler via CLI still log fails and responses in the normal way?

Yes, status updates and logging are the same—the only real difference is how the process is triggered.

> Any more specifics on the post url?

It should have this sort of structure (and will be a `POST` request):

`https://example.com/wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=a1b2c3d4e5`

**Comment by jacobmasters** [comment]
ok, i think we're going to use the CLI method, which would involve calling wp cli via a regular server cron job i think, like:

**** /usr/local/bin/wp --path=/home/jetpackaoi/public_html action-scheduler run --allow-root

Does that look correct?

**Comment by barryhughes** [comment]
The WP command itself looks good to me, though it is usually good to avoid using `--allow-root` (if you can) and normally the cron expression itself would have 5 elements (may be different on your system, though).
